packages/commercetools/theme/components/AppHeader.vue

```diff
@@ -0,0 +1,245 @@
+          @focus="isSearchOpen = true"
+          @keydown.esc="closeSearch"
+          v-click-outside="closeSearch"
+        >
+          <template #icon>
+            <SfButton
+              v-if="!!term"
+              class="sf-search-bar__button sf-button--pure"
+              @click="closeOrFocusSearchBar"
+            >
+              <span class="sf-search-bar__icon">
+                <SfIcon color="var(--c-text)" size="18px" icon="cross" />
+              </span>
+            </SfButton>
+            <SfButton
+              v-else
+              class="sf-search-bar__button sf-button--pure"
+              @click="isSearchOpen ? isSearchOpen = false : isSearchOpen = true"
+            >
+              <span class="sf-search-bar__icon">
+                <SfIcon color="var(--c-text)" size="20px" icon="search" />
+              </span>
+            </SfButton>
+          </template>
+        </SfSearchBar>
+      </template>
+    </SfHeader>
+    <SearchResults :visible="isSearchOpen" :result="result" @close="closeSearch" @removeSearchResults="removeSearchResults" />
+    <SfOverlay :visible="isSearchOpen" />
+  </div>
+</template>
+
+<script>
+import { SfHeader, SfImage, SfIcon, SfButton, SfBadge, SfSearchBar, SfOverlay, SfMenuItem, SfLink } from '@storefront-ui/vue';
+import { useUiState } from '~/composables';
+import { useCart, useUser, cartGetters } from '@vue-storefront/commercetools';
+import { computed, ref, onBeforeUnmount, watch } from '@vue/composition-api';
+import { useUiHelpers } from '~/composables';
+import StoreLocaleSelector from './StoreLocaleSelector';
+import SearchResults from '~/components/SearchResults';
+import { clickOutside } from '@storefront-ui/vue/src/utilities/directives/click-outside/click-outside-directive.js';
+import {
+  mapMobileObserver,
+  unMapMobileObserver
+} from '@storefront-ui/vue/src/utilities/mobile-observer.js';
+import debounce from 'lodash.debounce';
+import mockedSearchProducts from '../mockedSearchProducts.json';
+
+export default {
+  components: {
+    SfHeader,
+    SfImage,
+    StoreLocaleSelector,
+    SfIcon,
+    SfButton,
+    SfBadge,
+    SfSearchBar,
+    SearchResults,
+    SfOverlay,
+    SfMenuItem,
+    SfLink
+  },
+  directives: { clickOutside },
+  setup(props, { root }) {
+    const { toggleCartSidebar, toggleWishlistSidebar, toggleLoginModal } = useUiState();
+    const { setTermForUrl, getFacetsFromURL } = useUiHelpers();
+    const { isAuthenticated, load: loadUser } = useUser();
+    const { cart } = useCart();
+    const term = ref(getFacetsFromURL().phrase);
+    const isSearchOpen = ref(false);
+    const searchBarRef = ref(null);
+    const result = ref(null);
+
+    const cartTotalItems = computed(() => {
+      const count = cartGetters.getTotalItems(cart.value);
+      return count ? count.toString() : null;
+    });
+
+    const accountIcon = computed(() => isAuthenticated.value ? 'profile_fill' : 'profile');
+
+    loadUser();
```

Comment:
Is there a reason this is not wrapped in `onSSR`?

Reply:
It was copy-pasted from `nuxt-theme-module` we just changed the `LocaleSelector` to `StoreLocaleSelector` which is our new component for selecting stores